test/integration_test.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+
+import pytest
+from hydra import compose, initialize
+from omegaconf import DictConfig, OmegaConf
+
```

Comment:
Maybe just add a comment here to explain the logic behind the way we are running one system over all envs and then all systems over one env to save time.

Reply:
We can also just add some logic that will randomly select an env to run all systems on and then also random select a system to run all envs on.